### ISSUE TITLE
crash on launch (Ubuntu 19.04)

### ISSUE THREAD

**Issue by fekir** [SFTtech/openage#1190]
I'm able to import the assets, and the game starts (click start game, everything seems to work)
````
...
INFO [py] asset conversion complete; asset version: 7
INFO launching engine with Path(Union(Directory(/home/xxx/Workspace/openage/cfg).root @ (b'cfg',), Directory(/home/xxx/Workspace/openage/assets).root @ (b'assets',)):) and fps limit 0
INFO SDL audio subsystems initialized
INFO Using audio device: default [freq=48000, format=32784, channels=2, samples=4096]
INFO Initialized SDL video subsystems.
INFO Compiled with Qt 5.12.2 and run with Qt 5.12.2
INFO loading configuration files...
INFO [py] loading config file [Union(Directory(/home/xxx/Workspace/openage/cfg).root @ (b'cfg',), Directory(/home/xxx/Workspace/openage/assets).root @ (b'assets',))]:cfg/keybinds.oac...
INFO Loading time [engine]: 0.292 s
INFO Loading time   [game]: 0.314 s
INFO [T1] Loading game specification files...
INFO [T1] Loaded multi-csv file: 45484 sub-files
INFO [T1] Loading textures...
 ERR Can not profile: gperftools is missing
INFO [T1] Loading sounds...
INFO [T1] Loading time  [data]: 5.120 s
INFO Gaia has civilisation Gaia
INFO Jonas has civilisation British
INFO Michael has civilisation French
INFO cya!
````

but after closing the window, and trying to execute the program again, without further modification, it crashes:

````
INFO [py] launching openage v0.4.0-71-g9740fcb7
INFO [py] compiled by GNU 8.3.0
INFO [py] running in DEVMODE
INFO launching engine with Path(Union(Directory(/home/xxx/Workspace/openage/cfg).root @ (b'cfg',), Directory(/home/xxx/Workspace/openage/assets).root @ (b'assets',)):) and fps limit 0
INFO SDL audio subsystems initialized
INFO Using audio device: default [freq=48000, format=32784, channels=2, samples=4096]
INFO Initialized SDL video subsystems.
INFO Compiled with Qt 5.12.2 and run with Qt 5.12.2
INFO loading configuration files...
INFO [py] loading config file [Union(Directory(/home/xxx/Workspace/openage/cfg).root @ (b'cfg',), Directory(/home/xxx/Workspace/openage/assets).root @ (b'assets',))]:cfg/keybinds.oac...
INFO Loading time [engine]: 0.271 s
INFO Loading time   [game]: 0.289 s

SIGSEGV

FATAL: terminate has been called

current stack:

Traceback (most recent call last):
  File ?, in __libc_start_main+0xeb [0x7f4fcaa90b6b]
  File ?, in ? [0x5577456df86d]
  File ?, in ? [0x5577456df69e]
  File ?, in PyModule_ExecDef+0x4a [0x7f4fcb06b6ca]
  File ?, in ? [0x5577456def02]
  File ?, in ? [0x5577456ddd2b]
  File ?, in ? [0x5577456ddb7c]
  File ?, in _PyEval_EvalFrameDefault+0x7b95 [0x7f4fcaead9b5]
  File ?, in ? [0x7f4fcaeb04d3]
  File ?, in _PyEval_EvalFrameDefault+0x8bde [0x7f4fcaeae9fe]
  File ?, in _PyCFunction_FastCallKeywords+0x25 [0x7f4fcb0aca05]
  File ?, in _PyMethodDef_RawFastCallKeywords+0x30b [0x7f4fcb0abcdb]
  File ?, in ? [0x7f4fc816d71b]
  File ?, in ? [0x7f4fc816cce9]
  File ?, in openage::run_game(openage::main_arguments const&)+0x523 [0x7f4fc72fda9f]
  File ?, in openage::Engine::run()+0x27 [0x7f4fc72ec41f]
  File ?, in openage::Engine::loop()+0x11b [0x7f4fc72ebd7b]
  File ?, in openage::gui::GUI::process_events()+0xd [0x7f4fc73c16ff]
  File ?, in qtsdl::GuiEventQueue::process_callbacks()+0xc [0x7f4fc73c6e02]
  File ?, in qtsdl::GuiEventQueueImpl::process_callbacks()+0x1c [0x7f4fc73cc83e]
  File ?, in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)+0xf4 [0x7f4fc532b0d4]
  File ?, in g_main_context_iteration+0x2c [0x7f4fc41fad1c]
  File ?, in ? [0x7f4fc41fac88]
  File ?, in g_main_context_dispatch+0x2ae [0x7f4fc41fa9ee]
  File ?, in ? [0x7f4fc532ba43]
  File ?, in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*)+0x1c7 [0x7f4fc52da927]
  File ?, in QCoreApplication::notifyInternal2(QObject*, QEvent*)+0x179 [0x7f4fc52d78e9]
  File ?, in QObject::event(QEvent*)+0xe2 [0x7f4fc5301ca2]
  File ?, in ? [0x7f4fc5dbd589]
  File ?, in QQmlComponentPrivate::loadUrl(QUrl const&, QQmlComponent::CompilationMode)+0x176 [0x7f4fc5dbcf36]
  File ?, in QQmlComponent::statusChanged(QQmlComponent::Status)+0x3e [0x7f4fc5db9a3e]
  File ?, in QMetaObject::activate(QObject*, int, int, void**)+0x8a3 [0x7f4fc5301563]
  File ?, in QtPrivate::QSlotObject<void (qtsdl::GuiSubtreeImpl::*)(QQmlComponent::Status), QtPrivate::List<QQmlComponent::Status>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*)+0x50 [0x7f4fc73d09c7]
  File ?, in qtsdl::GuiSubtreeImpl::component_status_changed(QQmlComponent::Status)+0x125 [0x7f4fc73d034b]
  File ?, in QQmlComponentPrivate::completeCreate()+0x42 [0x7f4fc5dba4e2]
  File ?, in QQmlComponentPrivate::complete(QQmlEnginePrivate*, QQmlComponentPrivate::ConstructionState*)+0x81 [0x7f4fc5dba3d1]
  File ?, in QQmlObjectCreator::finalize(QQmlInstantiationInterrupt&)+0x114 [0x7f4fc5e35f94]
  File ?, in QQmlBinding::update(QFlags<QQmlPropertyData::WriteFlag>)+0x213 [0x7f4fc5e28c53]
  File ?, in ? [0x7f4fc5e2c372]
  File ?, in QQmlBinding::evaluate(bool*)+0xdf [0x7f4fc5e279cf]
  File ?, in QQmlJavaScriptExpression::evaluate(QV4::CallData*, bool*)+0x1d7 [0x7f4fc5e22d77]
  File ?, in QV4::Function::call(QV4::Value const*, QV4::Value const*, int, QV4::ExecutionContext const*)+0x16b [0x7f4fc5cad5eb]
  File ?, in ? [0x7f4fc5d1c297]
  File ?, in ? [0x7f4fc5d17dc4]
  File ?, in QV4::Runtime::method_loadProperty(QV4::ExecutionEngine*, QV4::Value const&, int)+0xc9 [0x7f4fc5d866d9]
  File ?, in QV4::QObjectWrapper::virtualGet(QV4::Managed const*, QV4::PropertyKey, QV4::Value const*, bool*)+0x9c [0x7f4fc5cfabcc]
  File ?, in QV4::QObjectWrapper::getQmlProperty(QQmlContextData*, QV4::String*, QV4::QObjectWrapper::RevisionMode, bool*, bool) const+0x4b3 [0x7f4fc5cfa913]
  File ?, in ? [0x7f4fc5cf9473]
  File ?, in ? [0x7f4fc74d8de9]
  File ?, in openage::gui::ActionModeLink::get_selection_size() const+0xc [0x7f4fc73a7994]
  File ?, in ? [0x7f4fcaaadf60]
  File ?, in ? [0x7f4fc73ddf5a]
  File ?, in ? [0x7f4fcacfa301]
  File ?, in ? [0x7f4fcacfa2b6]
  File ?, in openage::error::terminate_handler()+0x19d [0x7f4fc73de1dd]

handing over to the system...

terminate called without an active exception
zsh: abort (core dumped)  ./run
````

Is it a know issue? What information can I provide?
Notice that deleting and importing the assets again permits me to start again the game, but closing and reopening it makes it crash again.

g++: `g++ (Ubuntu 8.3.0-6ubuntu1) 8.3.0`
python3: `Python 3.7.3`

**Comment by heinezen** [comment]
Looks like this was introduced by #1154 . `get_selection_size()` might be called too soon. @MaanooAk

**Comment by DrissiReda** [comment]
Same issue on Debian buster, does this happen to everyone?

**Comment by heinezen** [comment]
Does the bug persist with Ubuntu 19.10?

This issue is probably related to #1197 . I indicated a possible explanation for the segfault there.

**Comment by heinezen** [comment]
Presumably fixed by #1199 . Please report back if the issue still persists.
